src/meshes/t8code_mesh.jl

```diff
@@ -108,6 +111,153 @@ Main mesh constructor for the `T8codeMesh` wrapping around a given t8code
+    Trixi.cmesh_ref = Ref(t8_cmesh_t())
+    t8_cmesh_init(Trixi.cmesh_ref)
+    cmesh = Trixi.cmesh_ref[]
+
+    # Use linear geometry for now. There is no real Lagrange geometry
+    # implementation yet in t8code.
+    Trixi.linear_geom = Trixi.t8_geometry_linear_new(2)
+    Trixi.linear_geom_ptr = pointer_from_objref(Ref(Trixi.linear_geom))
+    t8_cmesh_register_geometry(cmesh, Trixi.linear_geom_ptr)
+
+    # Determine element class.
+    eclass = ndims > 2 ? T8_ECLASS_HEX : T8_ECLASS_QUAD
+
+    # Store element vertices inside the cmesh.
+    N = length(nodes)
+    vertices = zeros(3 * 2^ndims) # quads/hexs only
+    for i in 1:ntrees
+        t8_cmesh_set_tree_class(cmesh, i - 1, eclass)
+
+        if ndims == 2
+            vertices[1] = tree_node_coordinates[1, 1, 1, i]
+            vertices[2] = tree_node_coordinates[2, 1, 1, i]
+
+            vertices[4] = tree_node_coordinates[1, N, 1, i]
+            vertices[5] = tree_node_coordinates[2, N, 1, i]
+
+            vertices[7] = tree_node_coordinates[1, 1, N, i]
+            vertices[8] = tree_node_coordinates[2, 1, N, i]
+
+            vertices[10] = tree_node_coordinates[1, N, N, i]
+            vertices[11] = tree_node_coordinates[2, N, N, i]
+        else
+            vertices[1] = tree_node_coordinates[1, 1, 1, 1, i]
+            vertices[2] = tree_node_coordinates[2, 1, 1, 1, i]
+            vertices[3] = tree_node_coordinates[3, 1, 1, 1, i]
+
+            vertices[4] = tree_node_coordinates[1, N, 1, 1, i]
+            vertices[5] = tree_node_coordinates[2, N, 1, 1, i]
+            vertices[6] = tree_node_coordinates[3, N, 1, 1, i]
+
+            vertices[7] = tree_node_coordinates[1, 1, N, 1, i]
+            vertices[8] = tree_node_coordinates[2, 1, N, 1, i]
+            vertices[9] = tree_node_coordinates[3, 1, N, 1, i]
+
+            vertices[10] = tree_node_coordinates[1, N, N, 1, i]
+            vertices[11] = tree_node_coordinates[2, N, N, 1, i]
+            vertices[12] = tree_node_coordinates[3, N, N, 1, i]
+
+            vertices[13] = tree_node_coordinates[1, 1, 1, N, i]
+            vertices[14] = tree_node_coordinates[2, 1, 1, N, i]
+            vertices[15] = tree_node_coordinates[3, 1, 1, N, i]
+
+            vertices[16] = tree_node_coordinates[1, N, 1, N, i]
+            vertices[17] = tree_node_coordinates[2, N, 1, N, i]
+            vertices[18] = tree_node_coordinates[3, N, 1, N, i]
+
+            vertices[19] = tree_node_coordinates[1, 1, N, N, i]
+            vertices[20] = tree_node_coordinates[2, 1, N, N, i]
+            vertices[21] = tree_node_coordinates[3, 1, N, N, i]
+
+            vertices[22] = tree_node_coordinates[1, N, N, N, i]
+            vertices[23] = tree_node_coordinates[2, N, N, N, i]
+            vertices[24] = tree_node_coordinates[3, N, N, N, i]
+        end
+
+        t8_cmesh_set_tree_vertices(cmesh, i - 1, vertices, 3)
+    end
+
+    # Connect the coarse mesh elements.
+    for i in 1:length(elemIDs)
```

Comment:
```suggestion
    for i in eachindex(elemIDs)
```
We usually prefer `eachindex` over `1:length`.